metagraph/src/cli/config/config.cpp

```diff
@@ -726,6 +726,8 @@ std::string Config::graphmode_to_string(DeBruijnGraph::Mode mode) {
             return "canonical";
         case DeBruijnGraph::PRIMARY:
             return "primary";
+        default:
+            throw std::runtime_error("Never happens");
```

Comment:
This isn't needed. Since it's an `enum`, adding or removing a value will lead to a compilation error anyway.

Reply:
gcc wasn't compiling it for some reason
https://github.com/ratschlab/metagraph/runs/1822432077

Reply:
It's because the function either needs to return a `DeBruijnGraph::Mode` type, or throw an exception. I'd move the `throw` to be outside the `switch` statement so we still get the desired behaviour. This way if we add a new mode, it'll fail to compile unless we add that mode here.

Reply:
I see what you mean